### ISSUE TITLE
DependencyCheck shows success message for publishing XML artifact to DependencyTrack but no results are shown 

### ISSUE THREAD

**Issue by quentinkhoo** [DependencyTrack/dependency-track#159]
Hi Steve,

So currently I have set up on my machine a Jenkins freestyle project which does an OWASP Dependency-Check against my local repository by following the guideline [here](https://docs.dependencytrack.org/integrations/jenkins/)

As stated in the title, the success message in the output console shows a success message as shown below 
![image](https://user-images.githubusercontent.com/33711159/40774628-ae0b944c-64f8-11e8-86fb-5ed274991418.png)

But on the DependencyTrack, I'm not seeing any results
![image](https://user-images.githubusercontent.com/33711159/40774643-b7c133c0-64f8-11e8-905e-61b0449d3e1a.png)

Are you able to advise why or what i might possibly be missing?

Hope to hear from you soon.

Edit:
If it matters, these are my log outputs on my side:
![image](https://user-images.githubusercontent.com/33711159/40776558-aa5a49c4-64fd-11e8-944a-33a39d0b8754.png)

**Comment by stevespringett** [comment]
The metrics do not show up right away in v3.0 - this has been corrected in v3.1.

However, it doesn't look like the report is being processed correctly. Are you uploading the XML report? If so, can you share it with me? (redact any confidential info if necessary before doing so)

**Comment by quentinkhoo** [comment]
Thank you for your response. I ran my Jenkins Build against a [WebGoat JAR File](https://github.com/WebGoat/WebGoat/releases/tag/v8.0.0.M17) and attached in this message is the generated XML report. Likewise I'm getting the same error as before.

![image](https://user-images.githubusercontent.com/33711159/40824325-528da124-65a6-11e8-8813-35d5c44a2fc6.png)

I hope to hear from you soon.

[dependency-check-report.zip](https://github.com/stevespringett/dependency-track/files/2061237/dependency-check-report.zip)

**Comment by adeline-chew-sp** [comment]
I'm having the same issue for the gradle plugin, the report is generated successfully but no report is shown in dependency-track.
```
[DependencyCheck] Collecting Dependency-Check analysis files...
[DependencyCheck] Searching for all files in /mnt/ubuntu/workspace/ADDY-dependency-check-atlas that match the pattern **/dependency-check-report.xml
[DependencyCheck] Parsing 1 file in /mnt/ubuntu/workspace/ADDY-dependency-check-atlas
[DependencyCheck] Successfully parsed file /mnt/ubuntu/workspace/ADDY-dependency-check-atlas/build/reports/dependency-check-report.xml with 34 unique warnings and 0 duplicates.
[DependencyCheck] Computing warning deltas based on reference build #44
[DependencyTrack] Publishing artifact to Dependency-Track
[DependencyTrack] The artifact was successfully published
```

![image](https://user-images.githubusercontent.com/10201631/40850062-3fe4f71a-6589-11e8-8711-a7088468a155.png)

**Comment by stevespringett** [comment]
I believe I know what the problem is. I haven't tested with v3.0.4 yet, but the file provided is properly parsed in v3.1.

In Dependency-Check XML reports, there an <identifier> type of maven for many dependencies. This format has always been (group:artifact:version) when the Maven plugin was responsible for creating the identifier. However, when the CLI or Gradle plugin was used, the parentheses are not in the identifier. This is actually a defect in the way the Dependency-Check Maven plugin constructs the identifier and this defect is part of the expected format for any system that parses the reports. This was solved in #157 so that the identifier can be parsed with and without the enclosing parentheses. The fix will be included in the upcoming v3.1 release this month.

That said, I did find the attached report useful for its use of related dependencies. Related dependencies are not presented that well in Dependency-Track and I'll likely use the file to help is parsing the data a bit better than it is now.

**Comment by stevespringett** [comment]
Duplicate/Related to #156

**Comment by stevespringett** [comment]
Included in v3.1.0. Closing.

**Comment by spawn-guy** [comment]
@stevespringett we are experiencing similar issues.
jenkins output
```
[Pipeline] dependencyTrackPublisher
[DependencyTrack] Publishing artifact to Dependency-Track
[DependencyTrack] The artifact was successfully published
```
log from running Docker
```
13:26:37.875 ERROR [ScanUploadProcessingTask] Error while processing scan result
java.lang.IllegalStateException: Error converting the PackageURL
	at org.owasp.dependencytrack.persistence.PackageURLStringConverter.convertToAttribute(PackageURLStringConverter.java:47)
	at org.owasp.dependencytrack.persistence.PackageURLStringConverter.convertToAttribute(PackageURLStringConverter.java:33)
	at org.datanucleus.api.jdo.JDOTypeConverter.toMemberType(JDOTypeConverter.java:59)
	at org.datanucleus.store.rdbms.mapping.java.TypeConverterMapping.getMemberValueFromDatastore(TypeConverterMapping.java:256)
	at org.datanucleus.store.rdbms.mapping.java.TypeConverterMapping.getObject(TypeConverterMapping.java:536)
	at org.datanucleus.store.rdbms.fieldmanager.ResultSetGetter.fetchObjectField(ResultSetGetter.java:181)
	at org.datanucleus.state.StateManagerImpl.replacingObjectField(StateManagerImpl.java:1953)
	at org.owasp.dependencytrack.model.Component.dnReplaceField(Component.java)
	at org.owasp.dependencytrack.model.Component.dnReplaceFields(Component.java)
	at org.datanucleus.state.StateManagerImpl.replaceFields(StateManagerImpl.java:4302)
	at org.datanucleus.state.StateManagerImpl.replaceFields(StateManagerImpl.java:4327)
	at org.datanucleus.store.rdbms.request.FetchRequest.execute(FetchRequest.java:403)
	at org.datanucleus.store.rdbms.RDBMSPersistenceHandler.fetchObject(RDBMSPersistenceHandler.java:316)
	at org.datanucleus.state.StateManagerImpl.loadFieldsFromDatastore(StateManagerImpl.java:1500)
	at org.datanucleus.state.StateManagerImpl.loadUnloadedFieldsInFetchPlan(StateManagerImpl.java:3879)
	at org.datanucleus.state.StateManagerImpl.isLoaded(StateManagerImpl.java:4094)
	at org.owasp.dependencytrack.model.Component.dnGetuuid(Component.java)
	at org.owasp.dependencytrack.model.Component.getUuid(Component.java:437)
	at org.owasp.dependencytrack.persistence.QueryManager.updateComponent(QueryManager.java:528)
	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.processDependency(ScanUploadProcessingTask.java:118)
	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.inform(ScanUploadProcessingTask.java:83)
	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:87)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.github.packageurl.MalformedPackageURLException: Invalid purl: Illegal character in opaque part at index 38: pkg:maven/org.fusesource.jansi/jansi-${platform}@1.7?type=xml
	at com.github.packageurl.PackageURL.parse(PackageURL.java:263)
	at com.github.packageurl.PackageURL.<init>(PackageURL.java:64)
	at org.owasp.dependencytrack.persistence.PackageURLStringConverter.convertToAttribute(PackageURLStringConverter.java:45)
	... 26 common frames omitted
```
first we tried Jenkins plugin for Android Application checking, but it produces "obfuscated" results (all modules are shown like numbers: 0,1,2,300,400 and 0.jar,1.jar,2.jar,300.jar,400.jar). 
Gradle plugin shows correct insights, and the vulnerability trend is correctly published to Jenkins-Dashboard (via Jenkins plugin)

**Comment by stevespringett** [comment]
@spawn-guy publishing artifacts is independent from the processing of those artifacts. The processing occurs async, so the expected behavior is to receive a 200 back when publishing artifacts.

But it seems you've hit an issue processing one of the artifacts. Can you provide the full dependency-check xml report please? If this is not possible, just include the dependencies that have org.fusesource.jansi.

**Comment by stevespringett** [comment]
@spawn-guy In addition to supplying the dependency-check xml report, I did some investigation.

A maven group or artifactId must conform to the following regex `[A-Za-z0-9_\\-.]+` as seen in https://github.com/apache/maven/blob/master/maven-model-builder/src/main/java/org/apache/maven/model/validation/DefaultModelValidator.java#L76. This requirement goes back to Maven 2 (and possibly earlier). As a result of this requirement, Dependency-Track does not perform any type of escaping or URL encoding of group/artifact values when constructing Package URLs.  Adding escaping/encoding to Dependency-Track would only mask a problem, not solve it.

So, I'd like to see the evidence in the Dependency-Check XML report to determine how ${platform} is being introduced. Is that something Dependency-Track is parsing incorrectly or using the wrong field to determine the artifact ID, or is Dependency-Check supplying Maven evidence that violates Maven's own requirements? TBD.

**Comment by stevespringett** [comment]
@spawn-guy FYI, I received the xml report and will investigate. Thanks.

**Comment by stevespringett** [comment]
@spawn-guy, I've verified that the issue is coming from invalid data in the dependency-check xml report. The 'fix' in dependency-track will simply skip these dependencies and will generate an error entry in the log file with information about what dependency violated the Maven spec.

I've reproduced the issue on a sample project and created an issue on the dependency-check project. https://github.com/jeremylong/DependencyCheck/issues/1393
